src/abi/interaction.local.net.spec.ts

```diff
@@ -276,19 +262,17 @@ describe("test smart contract interactor", function () {
         const parser = new SmartContractTransactionsOutcomeParser({ abi: abiRegistry });
 
         const bytecode = await promises.readFile("src/testdata/counter.wasm");
+        alice.nonce = (await provider.getAccount(alice.address)).nonce;
 
         const deployTransaction = factory.createTransactionForDeploy(alice.address, {
             bytecode: bytecode,
             gasLimit: 3000000n,
         });
-        deployTransaction.nonce = BigInt(alice.account.nonce.valueOf());
+        deployTransaction.nonce = alice.getNonceThenIncrement();
 
-        deployTransaction.signature = await alice.signer.sign(
-            Buffer.from(transactionComputer.computeBytesForSigning(deployTransaction)),
-        );
+        deployTransaction.signature = alice.signTransaction(deployTransaction);
 
-        const contractAddress = SmartContract.computeAddress(alice.address, alice.account.nonce);
-        alice.account.incrementNonce();
+        const contractAddress = SmartContract.computeAddress(alice.address, alice.nonce);
```

Comment:
if you use `getNonceThenIncrement()` and then try to compute the contract's address using `alice.nonce` you'll have a higher nonce than the one on the transaction, thus computing an invalid address.